src/reflect/scala/reflect/internal/tpe/TypeComparers.scala

```diff
@@ -365,9 +365,23 @@ trait TypeComparers {
 
   // @assume tp1.isHigherKinded || tp2.isHigherKinded
   def isHKSubType(tp1: Type, tp2: Type, depth: Depth): Boolean = {
-    def isSub(ntp1: Type, ntp2: Type) = (ntp1.withoutAnnotations, ntp2.withoutAnnotations) match {
+    def isSub(tp1: Type, tp2: Type) = (tp1.withoutAnnotations, tp2.withoutAnnotations) match {
       case (TypeRef(_, AnyClass, _), _)                                     => false                    // avoid some warnings when Nothing/Any are on the other side
       case (_, TypeRef(_, NothingClass, _))                                 => false
```

Comment:
Type aliases for `Any` and `Nothing` would evade the short circuit now that the caller doesn't normalize.

Reply:
Thanks, will move these 2 lines down to `isSub2`.